### PR TITLE
Add the GET API and getByDocumentId function in Back-End

### LINKED ISSUE #3: Create GET request to retrieve form data to pass data into middleware

### PR DESCRIPTION
**Pull request by ChaosZhai**
Try to support displaying scores. Add get function that responses to GET requests in ScoreResource.java, and getByDocumentId in ScoreDao.java. Resolves #3

### CODE REVIEW

File: docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java [hunk 1]
```diff
@@ -32,4 +37,35 @@ public String create(Score score, String userId) {
...
+
+    /**
+     * Get all scores on a document.
+     * 
+     * @param documentId Document ID
+     * @return List of scores
+     */
+    public List<ScoreDto> getByDocumentId(String documentId) {
+        EntityManager em = ThreadLocalContext.get().getEntityManager();
+        StringBuilder sb = new StringBuilder("select s.SCORE_ID_C, s.SCORE_SKILLS_C, s.SCORE_EXPERIENCE_C, s.SCORE_TRANSCRIPTGPA_C, SCORE_MATCH_C, SCORE_CREATEDATE_D from T_SCORE c");
```
Comment (joannetsaii) [inline docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java]: Please fix the typo at the end (should be "s"), and add the initial aliases to all the columns.

File: docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java [hunk 2]
```diff
@@ -32,4 +37,35 @@ public String create(Score score, String userId) {
...
+        
+        List<ScoreDto> scoreDtoList = new ArrayList<>();
+        for (Object[] o : l) {
+            int i = 0;
+            ScoreDto scoreDto = new ScoreDto();
+            scoreDto.setId((String) o[i++]);
+            scoreDto.setSkills((String) o[i++]);
+            scoreDto.setExperience((String) o[i++]);
+            scoreDto.setTranscriptGpa((String) o[i++]);
+            scoreDto.setMatch((String) o[i++]);
```
Comment (joannetsaii) [inline docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java:65]: Could you confirm if the scores (skills, experience, transcriptGpa, match) should be Integer or String type? The set functions return Integer types but I'm not sure if the `(String)` is casting the values to String.

File: docs-web/src/main/java/com/sismics/docs/rest/resource/ScoreResource.java [hunk 3]
```diff
@@ -97,4 +101,61 @@ public Response add(
...
+     * @apiPermission none
+     * @apiVersion 1.5.0
+     *
+     * @param documentId DocumentID
+     * @return Response
+     */
+    @GET
+    @Path("{documentId: [a-z0-9\\-]+}")
+    public Response get(@PathParam("documentId") String documentId,
+            @QueryParam("share") String shareId) {
```
Comment (joannetsaii) [inline docs-web/src/main/java/com/sismics/docs/rest/resource/ScoreResource.java:131]: Could you briefly explain what the "share" query param is used for and where the shareId will be retrieved from?

**Review by joannetsaii — CHANGES_REQUESTED** [review]

File: docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java (hunk 1 above)
Comment (ChaosZhai) [reply docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java]: Fixed in the recent commit!

File: docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java (hunk 2 above)
Comment (ChaosZhai) [reply docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java:65]: The set functions take in String parameters here and cast to integers.

File: docs-web/src/main/java/com/sismics/docs/rest/resource/ScoreResource.java (hunk 3 above)
Comment (ChaosZhai) [reply docs-web/src/main/java/com/sismics/docs/rest/resource/ScoreResource.java:131]: Not sure, probably an already bound environment variable, and only used for checking permission. Refer to CommentResource.java for its usage.

**Review by joannetsaii — APPROVED** [review]

**Pull request merged**